src/sharing/ShareToStream.js

```diff
@@ -149,13 +149,20 @@ class ShareToStream extends React.Component<Props, State> {
     const { stream, topic, message } = this.state;
     const { sharedData } = this.props.route.params;
 
-    if (sharedData.type !== 'text') {
+    if (!sharedData.isText) {
       return stream !== '' && topic !== '';
     }
 
     return stream !== '' && topic !== '' && message !== '';
   };
 
+  renderItem = ({ item, index, separators }) =>
+    item.type.startsWith('image') ? (
+      <Image source={{ uri: item.url }} style={styles.imagePreview} />
+    ) : (
+      <></>
+    );
+
```

Comment:
Hmm, it's a shame to have to be making all these changes twice. And it makes it likely that as we keep making changes, things will unintentionally diverge.

Would you try to deduplicate some of this code, in commits that come before the ones where you extend it?

It might be as simple as making a component with a name like `SharedContentPreview`, which would contain (initially, before your change to handle multiple files) just the `<Image … />` line and the conditional on `sharedData.type` that controls it.

Basically any code you're going to add more logic to, and that's already duplicated, would be good to deduplicate first so you can then add the logic only once. (Ideally it should have been deduplicated in the first place; oh well :shrug: )

Reply:
I've made a `ShareWrapper` method, that deduplicates this and a lot more. Initially I went with just `SharedContentPreview` but there was a lots of de-duplication potential in both the screens. I wonder if this can have any downside though :thinking:.